dot/parachain/types/bitvec.go

```diff
@@ -82,6 +71,46 @@ func bytesToBits(b []byte, size uint) []bool {
 			bits = append(bits, bit)
 		}
 	}
+	bv.bits = bits
+}
+
+// MarshalSCALE fulfils the SCALE interface for encoding
+func (bv BitVec) MarshalSCALE() ([]byte, error) {
```

Comment:
make `bv` a pointer to consistent with other methods

Reply:
I can't make this method a pointer receiver.  If I make it a pointer receiver, then the `BitVec` type no longer implements the `scale.Marshaler` interface, only `*BitVec` would.  By not making it a pointer receiver both `BitVec` and `*BitVec` will implement the `scale.Marshaler` interface.  This is something I discovered when introducing the `scale.Marshaler` and `scale.Unmarshaler` interfaces.